pkg/kubelet/apis/config/validation/validation_others.go

```diff
@@ -20,10 +20,16 @@ limitations under the License.
 package validation
 
 import (
+	"fmt"
+
 	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
 )
 
 // validateKubeletOSConfiguration validates os specific kubelet configuration and returns an error if it is invalid.
 func validateKubeletOSConfiguration(kc *kubeletconfig.KubeletConfiguration) error {
+	if kc.SingleProcessOOMKill != nil {
+		return fmt.Errorf("invalid configuration: singleProcessOOMKill is only supported on linux")
```

Comment:
this will cause the same issue kubeadm ran into in https://github.com/kubernetes/kubeadm/issues/2967 where configs made centrally on a linux machine will fail when distributed to windows nodes

Reply:
@marosset this is the comment I was asking about. Thank you for confirming that this is fine

Reply:
The windows PR job that ran against this PR uses kubeadm and the Windows successfully came up.
If we think there is a risk of mutations getting applied to the kubelet config that would cause the value to be specified for Windows then we should just warn (like how kubernetes/kubeadm#2967 was resolved).

Reply:
Should we address this? 
- just warning for windows.

/cc @neolit123

Reply:
Done in the validation for windows.
https://github.com/kubernetes/kubernetes/pull/126096/files#diff-190c7fa882f6287a4555d876a2c5072ba794ab799551b403f6db86ba74b87215R40-R43